diff --git a/node_modules/vue-bundle-renderer/dist/runtime.cjs b/node_modules/vue-bundle-renderer/dist/runtime.cjs
index f105c86..4f378c4 100644
--- a/node_modules/vue-bundle-renderer/dist/runtime.cjs
+++ b/node_modules/vue-bundle-renderer/dist/runtime.cjs
@@ -150,8 +150,8 @@ function getPreloadLinks(ssrContext, rendererContext) {
 function getPrefetchLinks(ssrContext, rendererContext) {
   const { prefetch } = getRequestDependencies(ssrContext, rendererContext);
   return Object.values(prefetch).map((resource) => ({
-    rel: "prefetch" + (resource.resourceType === "style" ? " stylesheet" : ""),
-    as: resource.resourceType !== "style" ? resource.resourceType : null,
+    rel: 'prefetch',
+    as: resource.resourceType,
     type: resource.mimeType ?? null,
     crossorigin: resource.resourceType === "font" || resource.module ? "" : null,
     href: rendererContext.buildAssetsURL(resource.file)
